tests/lib/rules/no-constant-condition.js

```diff
@@ -51,6 +58,12 @@ ruleTester.run("no-constant-condition", rule, {
         { code: "while(~!0);", errors: [{ message: "Unexpected constant condition.", type: "WhileStatement"}] },
         { code: "while(x = 1);", errors: [{ message: "Unexpected constant condition.", type: "WhileStatement"}] },
         { code: "while(function(){});", errors: [{ message: "Unexpected constant condition.", type: "WhileStatement"}] },
-        { code: "while(() => {});", parserOptions: { ecmaVersion: 6 }, errors: [{ message: "Unexpected constant condition.", type: "WhileStatement"}] }
+        { code: "while(() => {});", parserOptions: { ecmaVersion: 6 }, errors: [{ message: "Unexpected constant condition.", type: "WhileStatement"}] },
+
+        // #5693
+        { code: "if(false && abc==='str'){}", errors: [{message: "Unexpected constant condition.", type: "IfStatement"}]},
+        { code: "if(false || abc==='str'){}", errors: [{message: "Unexpected constant condition.", type: "IfStatement"}]}
```

Comment:
This isn't constant. The right side of the `||` will still be evaluated, unlike with `&&`